runtime/compiler/env/j9method.cpp

```diff
@@ -4522,6 +4522,236 @@ void TR_ResolvedJ9Method::construct()
             if (!strncmp(name, "invokeExact_thunkArchetype_", 27))
                setRecognizedMethodInfo(TR::java_lang_invoke_InterfaceHandle_invokeExact);
             }
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+         else if ((classNameLen == 40 && !strncmp(className, "java/lang/invoke/VarHandleBooleans$Array", 40)))
```

Comment:
These classes seem to have other methods in addition to the ones that we're trying to detect here. It's not obvious to me that those other methods don't (or won't) lead to an `Unsafe` call in some (transitive) callee, and I'm concerned about drawing incorrect conclusions about such a call if one were to occur. Could we restrict the recognition here to try to limit it to just the actual VH access operations? e.g. maybe recognize only static methods with certain names and first parameter of type `VarHandle`

Reply:
I added a check to address this.